src/daemon/default_vm_image_vault.cpp

```diff
@@ -251,6 +252,12 @@ mp::DefaultVMImageVault::DefaultVMImageVault(std::vector<VMImageHost*> image_hos
       prepared_image_records{load_db(cache_dir.filePath(image_db_name))},
       instance_image_records{load_db(data_dir.filePath(instance_db_name))}
 {
+    // TODO: Remove after Multipass 1.17
+    // The OS field will be unpopulated for existing images in the vault. As of 1.16, the only
+    // images in the prepared image records are Ubuntu images. Therefore, we can safely assume that
+    // if the OS field is empty, it was a previously existing Ubuntu cloud image. The same can be
+    // said for instance image records with instances created with the Alias Query::Type.
+    amend_db();
```

Comment:
I haven't looked at the vault code yet... do we tag the vault data with a version identifier? From this comment, I'm guessing we don't. It might be a good idea to add eventually (i.e. not in this PR), since then it's easy to migrate the vault from one schema to another.

This case is easy enough since it's just adding a missing field, but there might be more complicated updates in the future.

Reply:
Yeah, a version key might be beneficial in the future.